### DISCUSSION TITLE
Dexcom One Bloototh connection problem with Nightly build 23rd Mar 2023

### DISCUSSION THREAD

**Discussion by dar-kod** [NightscoutFoundation/xDrip#2748] [General]
Phone: Poco F3
Android version 13, MIUI 14.0.4

Yesterday I started with the first Dexcom One transmitter. During the night it was disconnected but not paired (how this was possible I do not know). I temporarily changed the transmitter ID to trigger a connection and paired it a second time. After 14 hours it was disconnected again but paired. In this case I did the following several times: clear cache, reboot, temporarily change transmitter ID. This worked for me on Android 12, MIUI 13. But now it did not help. The system status screenshot is attached. I wrote to Navid and he suggested to install nightly build 12 Jan 2023. I agreed, installed, changed ID and was about to change ID back to real ID when a window appeared asking me to download the keks plugin needed for the Dexcom One transmitter. I downloaded it and got readings immediately.

![1679789093207](https://user-images.githubusercontent.com/24446806/227748136-44f2d9cd-155a-45e7-8297-d429f0a3ffc0.jpg)

**Comment by Navid200** [comment]
Thanks for reporting this.

If you get a chance to try the other xDrip releases, it can be very helpful to us.
For example, if you try the release of March 15, 2023 and let us know if that one also fails or not, it can be very helpful.
Assuming that version works for you know.  it will be very useful if you try the latest release gain without an uninstall.  Does the last release work for you now as well?

 Of course controlling your diabetes is the most important thing.  So, if you prefer not to perform these experiments, it's understandable.
The fact is that not many developers (if any at all) may have access to Dexcom One, I don't.  I would perform those experiments if I did.

**Comment by jamorham** [comment]
@dar-kod how are things now? Can you explain what you mean by things working with android 12 but not with android 13. Did you upgrade your phone recently?

**Comment by dar-kod** [comment]
@jamorham It may have disconnected once or twice, but after a long time and temporarily changing the transmitter ID, the transmitter would reconnect. I was on Android 12 and upgraded to 13 and used it with the old G6 transmitter for about a week without any problems. Disconnections happen a lot at first, then rarely, and then they stop. It's probably Bluetooth learning about connections. As for previous disconnections with MIUI 13, Android 12, they seem to have been caused by using the Cleaner application, which I no longer use.

@Navid200 I just tried the 15 March 2023 release and it fails too. In "System Status" -> "Brain State" something like "Power Error 4" was written. I'm now trying the 6 March 2023 release and it seems to be working. I'll let you know if it does.

**Reply by rudysroost** [reply]
Android 13 did not work until the December update. Check for an Android update. I have Pixel 7 with the 13 March update and it is all working for me with the latest xDrip nightly build.

**Reply by dar-kod** [reply]
@rudysroost You mean everything works with Dexcom One?

**Reply by dar-kod** [reply]
@Navid200 I had one disconnection, which I consider normal. It was solved by changing the ID.

On another phone, Poco M4 Pro, the March 15, 2023 release works without any disconnections. With Poco F3 I'm now trying [2023.03.28b](https://github.com/NightscoutFoundation/xDrip/tree/2023.03.28b).

**Reply by dar-kod** [reply]
@Navid200 The [2023.03.28b](https://github.com/NightscoutFoundation/xDrip/tree/2023.03.28b) release was working for a few hours and I had to switch back to the 6 March 2023 release.

**Reply by Navid200** [reply]
Thanks for testing and reporting.
As long as you don't compromise your diabetes control, the next thing that may help is if you try the March 15, 2023 release.

**Reply by dar-kod** [reply]
I wrote about March 15, 2023 release at https://github.com/NightscoutFoundation/xDrip/discussions/2748#discussioncomment-5446180 : "I just tried the 15 March 2023 release and it fails too. In "System Status" -> "Brain State" something like "Power Error 4" was written."

**Reply by Navid200** [reply]
When you install a version older than March 15,  a plugin is installed.

After that, I wonder if installing the March 15 may work.

**Reply by dar-kod** [reply]
I have installed over the existing version. Should I export the settings, uninstall and try again with the March 15?

**Reply by Navid200** [reply]
No, absolutely not.  I'm just wondering if you now install the March 15 version over your existing March 6, does it still fail or not.

**Reply by dar-kod** [reply]
Okay. I tried again. No, it still fails. In "System Status" -> "G5/G6 Status" -> "Brain State:" these messages are constantly changing: 

```
Cheching Auth
Errors: 24
```

```
Power connect
Errors: 25
```

**Reply by Navid200** [reply]
Thanks

**Comment by anamljet** [comment]
I downloded the March 15, 2023 and did not have any connection.
After I tried March 23, 2023
I can connect, it works for few hours and it stops ( happend 3 times in 3 days)
Dexcome One and Redmi note 11 phone

**Comment by dar-kod** [comment]
@Navid200 It's been 2 months since the nightly release for Dexcom One is not working for me. Can you tell me if this is going to be fixed soon, because other issues with the March 6th release are causing me problems as well, and I want to be sure that if the issue with AAPS sending data to xDrip+ is fixed, I can use the latest nightly release.

**Reply by Navid200** [reply]
You should be able to use the latest Nightly.
If you cannot, it is a coincidence.  
We have no reason to believe there is anything wrong with xDrip for C devices.  
There are phones that insist to put apps that are in the background to sleep.  For every phone, there may be different needs to remedy that.

This page is updated frequently to address everything new that I find:
https://navid200.github.io/xDrip/docs/Intermittent.html

**Comment by dar-kod** [comment]
Yes, you are right. Thanks for the page. I finely found how to switch off "MIUI Optimizations" mentioned at https://dontkillmyapp.com/xiaomi. There it is not mentioned that "MIUI Optimizations" can be hidden in "Developer Options" and there you have to tap "Reset to default values" until "MIUI Optimizations" appears. After unchecking it, the latest nightly install works fine.

I also see that there is an app "DontKillMyApp: Make apps work" https://play.google.com/store/apps/details?id=com.urbandroid.dontkillmyapp which can be used to check how well everything is set up for a particular phone. I will let it run overnight.
